tests/performance/CatalogClientTest.cpp

```diff
@@ -0,0 +1,218 @@
+    client_settings.authentication_settings = auth_settings;
+    client_settings.task_scheduler = task_scheduler;
+    client_settings.network_request_handler = s_network;
+    client_settings.proxy_settings = GetLocalhostProxySettings();
+
+    return client_settings;
+  }
+
+ protected:
+  static std::shared_ptr<olp::http::Network> s_network;
+};
+
+std::shared_ptr<olp::http::Network> CatalogClientTest::s_network;
+
+namespace {
+
+void ClientThread(
+    const std::uint8_t client_id,
+    std::shared_ptr<olp::dataservice::read::CatalogClient> service_client,
+    std::string layer_id, std::chrono::milliseconds sleep_interval,
+    std::chrono::seconds runtime,
+    std::atomic<olp::http::RequestId>& request_counter) {
+  std::atomic_int success_responses{0};
+  std::atomic_int failed_responses{0};
+
+  std::mutex errors_mutex;
+  std::map<int, int> errors;
+
+  const auto end_timestamp = std::chrono::steady_clock::now() + runtime;
+
+  while (end_timestamp > std::chrono::steady_clock::now()) {
+    const auto partition_id = request_counter.fetch_add(1);
+
+    auto request = olp::dataservice::read::DataRequest()
+                       .WithLayerId(layer_id)
+                       .WithPartitionId(std::to_string(partition_id));
+
+    service_client->GetData(
+        request, [&](olp::dataservice::read::DataResponse response) {
+          if (response.IsSuccessful()) {
+            success_responses.fetch_add(1);
+          } else {
+            failed_responses.fetch_add(1);
+
+            // Collect errors information
+            std::lock_guard<std::mutex> lock(errors_mutex);
+            const auto& error = response.GetError();
+            const auto error_code = static_cast<int>(error.GetErrorCode());
+            const auto error_it = errors.find(error_code);
+            if (error_it != errors.end()) {
+              error_it->second++;
+            } else {
+              errors[error_code] = 0;
+            }
+          }
+        });
+
+    std::this_thread::sleep_for(sleep_interval);
+  }
+
+  OLP_SDK_LOG_CRITICAL_INFO_F(
+      "ClientThread",
+      "Client %d finished, succeed responses %d, failed responses %d",
+      client_id, success_responses.load(), failed_responses.load());
+
+  for (const auto& error : errors) {
+    OLP_SDK_LOG_CRITICAL_INFO_F("ClientThread",
+                                "Client %d, error %d - count %d", client_id,
+                                error.first, error.second);
+  }
+}
+}  // namespace
+
+/*
+ * Test performs N requests per second for M duration. All requests are unique.
+ * Total requests number calculated as:
+ *
+ * total_count = requests_per_second * calling_thread_count
+ *
+ * By default, SDK initialized in-memory cache with 1 MB capacity.
+ * To run the test, you need to start a local OLP mock server first.
+ * Valgrind, heaptrack, other tools are used to collect the output.
+ */
+TEST_P(CatalogClientTest, ReadNPartitions_VersionedLayer) {
```

Comment:
underscores in test names are dangerous
```suggestion
TEST_P(CatalogClientTest, ReadNPartitionsFromVersionedLayer) {
```